libs/api/domains/criminal-record/src/lib/graphql/main.resolver.ts

```diff
@@ -0,0 +1,22 @@
+import { Args, Query, Resolver } from '@nestjs/graphql'
+import { UseGuards } from '@nestjs/common'
+import { IdsUserGuard, ScopesGuard } from '@island.is/auth-nest-tools'
+import { CriminalRecordService } from '../criminalRecord.service'
+import { CriminalRecord } from './models'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Resolver()
+export class MainResolver {
+  constructor(private readonly criminalRecordService: CriminalRecordService) {}
+
+  //Note: not being used
+  @Query(() => CriminalRecord)
+  async getCriminalRecord(@Args('ssn') ssn: string) {
+    return await this.criminalRecordService.getCriminalRecord(ssn)
+  }
+
+  @Query(() => Boolean)
+  async checkCriminalRecord(@Args('ssn') ssn: string) {
+    return await this.criminalRecordService.checkCriminalRecord(ssn)
```

Comment:
Both of these queries look dangerous.
By implementing it like this you're exposing every single criminal record to users that have a valid login. It's most always incorrect to pass in the nationalId as an argument.
In addition, this is not following the naming conventions for GraphQL, ref: https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions

Arguments should be input objects and names `input`. The name of the queries should be the domain itself:
`CriminalRecord`. And this should most likely use the CurrentUser as the nationalId in order to prevent exposing the pre-mentioned danger ^.